vpr/src/route/segment_stats.cpp

```diff
@@ -41,25 +39,28 @@ void get_segment_usage_stats(std::vector<t_segment_inf>& segment_inf) {
         max_segment_name_length = std::max(max_segment_name_length, static_cast<int>(segment_inf[seg_type].name.size()));
     }
 
-    seg_occ_by_length = new int[max_segment_length + 1];
-    seg_cap_by_length = new int[max_segment_length + 1];
+    std::vector<int> seg_occ_by_length(max_segment_length+1, 0);
```

Comment:
Could probably get rid of this entirely, and compute from the ax and ay data structures.

Reply:
I think we can get rid of the total (x+y) usage by length.
Can keep the total (x+y) usage by segment type (aggregates data for electrically/topologically equivalent wires already).